src/prx/test/test_main.py

```diff
@@ -68,7 +71,98 @@ def test_prx_function_call_with_csv_output(input_for_test):
     assert expected_prx_file.exists()
     df = pd.read_csv(expected_prx_file, comment="#")
     assert not df.empty
+    assert helpers.is_sorted(df.time_of_reception_in_receiver_time)
     # Elevation sanity check
     assert (
         df[(df.prn == 14) & (df.constellation == "C")].elevation_deg - 34.86
     ).abs().max() < 0.3
+
+
+def parse_prx_file_metadata(prx_file: Path):
+    with open(prx_file, "r") as f:
+        metadata = json.loads(f.readline().replace("# ", ""))
+    return metadata
+
+
+def run_rinex_through_prx(rinex_obs_file: Path):
+    main.process(observation_file_path=rinex_obs_file, output_format="csv")
+    expected_prx_file = Path(
+        str(rinex_obs_file).replace("crx.gz", constants.cPrxCsvFileExtension)
+    )
+    assert expected_prx_file.exists()
+    metadata = parse_prx_file_metadata(expected_prx_file)
+    records = pd.read_csv(expected_prx_file, comment="#")
+    assert not records.empty
+    assert metadata
+    records.group_delay_m = records.group_delay_m.fillna(0)
+    records = records[records.C_obs.notna() & records.x_m.notna()]
+    return records, metadata
+
+
+def spp_lsq(df):
```

Comment:
Those functions (csv parser and spp_lsq) are quite useful, it kind of pains me to see them as additional functions in a `test_*.py` file.

Although they are not at the heart of **prx**, we may still put them forward in a separate `prx_use_illustration.py`, or something like that?

Reply:
Good call - I moved them into a separate `user.py` module :)